packages/components/src/powered-by/index.tsx

```diff
@@ -0,0 +1,82 @@
+import './style.scss';
+
+import colorStudio from '@automattic/color-studio';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import React from 'react';
+import { JetpackLogo } from '../logos/jetpack-logo';
+import { WooLogo } from '../logos/woo-logo';
+import WordPressWordmark from '../wordpress-wordmark';
+
+const PALETTE = colorStudio.colors;
+const COLOR_WPCOM_BLUE = PALETTE[ 'Blue 50' ];
+const COLOR_WHITE = PALETTE[ 'White' ];
+const COLOR_BLACK = PALETTE[ 'Black' ];
+
+export type PoweredByBrand = 'jetpack' | 'woocommerce' | 'wpcom';
+export type PoweredByVariant = 'color' | 'black' | 'white';
+
+interface PoweredByProps {
+	brand: PoweredByBrand;
+	colorVariant?: PoweredByVariant;
+	className?: string;
+}
+
+const getBrandName = ( brand: PoweredByBrand ) => {
+	switch ( brand ) {
+		case 'jetpack':
+			return 'Jetpack';
+		case 'woocommerce':
+			return 'WooCommerce';
+		case 'wpcom':
+			return 'WordPress.com';
+	}
+};
+
+const getWordPressWordmarkColor = ( colorVariant: PoweredByVariant ) => {
+	switch ( colorVariant ) {
+		case 'white':
+			return COLOR_WHITE;
+		case 'black':
+			return COLOR_BLACK;
+		default:
+			return COLOR_WPCOM_BLUE;
+	}
+};
+
+const PoweredBy = ( { brand, colorVariant = 'color', className }: PoweredByProps ) => {
+	let LogoComponent: React.ReactNode = null;
+
+	switch ( brand ) {
+		case 'jetpack':
+			LogoComponent = <JetpackLogo colorVariant={ colorVariant } size={ 25 } full />;
+			break;
+		case 'woocommerce':
+			LogoComponent = <WooLogo colorVariant={ colorVariant } height={ 25 } width={ 66 } />;
+			break;
+		case 'wpcom':
+			LogoComponent = (
+				<WordPressWordmark
+					color={ getWordPressWordmarkColor( colorVariant ) }
+					size={ { height: 25, width: 'auto' } }
+				/>
+			);
+			break;
+	}
+
+	return (
+		<p className={ clsx( 'powered-by', className, `is-${ brand }`, `is-${ colorVariant }` ) }>
+			{ createInterpolateElement( __( '<text>Powered by</text> <logo/>', 'calypso' ), {
```

Comment:
Should this text be translatable?

Reply:
I don't see why it shouldn't be

Reply:
though it doesn't need the 'calypso' there 🤔

Reply:
IMHO translating branding is more complicated than simply translating a text. 

For example, in Stats footer we right now display "Propulsate de Jetpack (logo)" for Romanian (it should be Propulsat de X, but that's not important). 

However, a literal translation in Romanian doesn't make much sense. Every Romanian site (or ads, or anything) I have seen uses the English version (Powered by X) - I haven't seen someone say "Propulsat de X". 

On a side note, in Romanian Propulsat is more used as "propelled", which is a bit funny for "Jetpack". 

What I'm trying to say is that branding might also be related to cultural usage(?). :)

Reply:
That sounds like it's maybe an issue with the Romanian localisation rather than an issue with the internationalisation.

If the norm in Romanian is to use the English phrase "powered by", then the Romanian translators can freely do that.

Reply:
If the context of in which the text is translated can make a difference (eg. WordPress.com, Jetpack, WooCommerce...), why hardcode `calypso`  though?

Reply:
I don't think the logo context does change how to translate the text.

The Calypso domain was a mistake, which I acknowledged elsewhere and then failed to actually fix. 😞